File: MQTTCSharpExample/Structs/MQTTSignalFrameData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MQTTCSharpExample
{
    public struct MQTTSignal
    {
        public string Timestamp;
        public string Name;
        public string Type;
        public string UnitX;
        public string UnitY;
        public string UnitZ;
        public ulong BlockSize;
        public double SamplingRate;
        public string WindowType;
        public string DisplayFormat;
    }

    public struct MQTTSignalFrameData
    {
        public MQTTSignal Signal;
        public double[] ValueX;
        public double[] ValueY;
        public double[] ValueZ;

        public int XSequenceType;//0-->linear,1-->log
        public double XStart;
        public double XDelta;
        public int XLength;
    }

    public struct MQTTSignalProperty
    {
        public string SignalName;
        public string PropertyName;
        public double Value;
        public string Unit;
    }

}

File: MQTTJavaDemoCode/JavaSwingDemoMqtt GUI app/MQTTCSharpProj/src/main/java/org/example/messages/MqttSignalFrameData.java
package org.example.messages;

import java.util.ArrayList;

public class MqttSignalFrameData {
    public Signals Signal;
    public ArrayList<Double> ValueX;
    public ArrayList<Double> ValueY;
    public ArrayList<Double> ValueZ;

    public MqttSignalFrameData(){

    }

    public MqttSignalFrameData(Signals signal, ArrayList<Double> valueX, ArrayList<Double> valueY, ArrayList<Double> valueZ) {
        Signal = signal;
        ValueX = valueX;
        ValueY = valueY;
        ValueZ = valueZ;
    }

    @Override
    public String toString() {
        return "MqttSignalFrameData{" +
                "Signal=" + Signal +
                ", ValueX=" + ValueX +
                ", ValueY=" + ValueY +
                ", ValueZ=" + ValueZ +
                '}';
    }
}
